### PR TITLE
Update `fav`/`unfav` feature to be more specific in developer guide

### LINKED ISSUE #263: Update `fav`/`unfav` feature in developer guide
- Make it more specific by using example command instead of variables
- Make the diagrams easier to read

### PR DESCRIPTION
**Pull request by tsulim**
Closes #263

### CODE REVIEW

File: docs/diagrams/FavActivityDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+if () then ([index is valid])
+    if () then ([Person has favourite as false])
+        :Construct a new Person object with favourite as true;
+        :Replace the original Person with the new one;
+        :Refresh the list of Person;
+    else ([else])
+        :Throws CommandException that Person has favourite as true;
+    endif
+else ([else])
+    :Throw CommandException that index is is invalid;
```
Comment (iynixil) [inline docs/diagrams/FavActivityDiagram.puml]: I think activity diagrams are not meant to have technical terms e.g `Throws CommandException`, perhaps `Show error message to user that person is already a favourite` is better?
on the same note,
- the "if" condition may be better phrased as `Person has favourite as false` -> `Person is not a favourite`
- line 13 `Construct a new Person object with favourite as true` -> `Create new Person with same values but is a favourite`

nit: line 15 `list of Person` -> `list of Persons`
nit: line 20 `index is is invalid` -> `index is invalid`

File: docs/diagrams/FavRefSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+box Model MODEL_COLOR_T1
+participant ":Person" as Person MODEL_COLOR
+end box
+
+FavCommand -> FavCommand1 : createFavPerson(personToFav)
+activate FavCommand
+activate FavCommand1
+
+FavCommand1 -> PersonUtil : createPersonWithFavouriteStatus(Person selectedPerson, Favourite favourite)
```
Comment (iynixil) [inline docs/diagrams/FavRefSequenceDiagram.puml]: I think the data type can be omitted, e.g `Person selectedPerson` -> `selectedPerson`

File: docs/diagrams/UnfavActivityDiagram.puml
Comment (iynixil) [inline docs/diagrams/UnfavActivityDiagram.puml:1]: same issue with `FavActivityDiagram.puml`

File: docs/diagrams/UnfavRefSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+box Model MODEL_COLOR_T1
+participant ":Person" as Person MODEL_COLOR
+end box
+
+
+UnfavCommand -> UnfavCommand1 : createUnfavPerson(personToUnfav)
+activate UnfavCommand1
+
+UnfavCommand1 -> PersonUtil : createPersonWithFavouriteStatus(Person selectedPerson, Favourite favourite)
```
Comment (iynixil) [inline docs/diagrams/UnfavRefSequenceDiagram.puml]: same issue with `FavRefSequenceDiagram.puml` about omitting data type

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -660,52 +705,43 @@ testers are expected to do more *exploratory* testing.
 1. Initial launch
 
    1. Download the jar file and copy into an empty folder
-
-   1. Double-click the jar file Expected: Shows the GUI with a set of sample contacts. The window size may not be optimum.
-
-1. Saving window preferences
+   2. Double-click the jar file Expected: Shows the GUI with a set of sample contacts. The window size may not be optimum.
```
Comment (iynixil) [inline docs/DeveloperGuide.md:712]: from line 708 onwards, i see that a blank line has been deleted between every numbered step, personally think it is good to have the blank line in between steps for clean visual.

note: My observation is having the blank line makes a difference in webpage but not sure about IDE, the difference between having blank line or not may not be shown in IDE

**Review by iynixil — COMMENTED** [review]
some clarifications and nits
note: remember to update PNGs after the changes have been made 🦺

File: docs/diagrams/FavRefSequenceDiagram.puml (hunk 2 above)
Comment (tsulim) [reply docs/diagrams/FavRefSequenceDiagram.puml]: Oh yes thank you for spotting it! I must have missed out when copying the function over...

File: docs/diagrams/FavActivityDiagram.puml (hunk 1 above)
Comment (tsulim) [reply docs/diagrams/FavActivityDiagram.puml]: I understand, thank you for pointing it out! With that said, I think `EditActivityDiagram` also contains those errors.

Comment (iynixil) [reply docs/diagrams/FavActivityDiagram.puml]: oh, will have to correct that in another PR, thanks for pointing out!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (tsulim) [reply docs/DeveloperGuide.md:712]: My apologies, I failed to realize that my IDE automatically removed the blank lines and changed the numbers. 😨
Will make amendments! 🥺

**Review by iynixil — APPROVED** [review]

**Pull request merged**